ddl/partition.go

```diff
@@ -1849,6 +1854,22 @@ func (w *worker) onDropTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (
 				return ver, errors.Trace(err)
 			}
 		}
+		if tblInfo.TiFlashReplica != nil {
+			// Remove the partitions
+			ids := tblInfo.TiFlashReplica.AvailablePartitionIDs
+			// Rarely called, so OK to take some time, to make it easy
+			for _, id := range physicalTableIDs {
+				for i, avail := range ids {
+					if id == avail {
+						tmp := ids[:i]
+						tmp = append(tmp, ids[i+1:]...)
+						ids = tmp
+						break
+					}
+				}
+			}
+			tblInfo.TiFlashReplica.AvailablePartitionIDs = ids
+		}
```

Comment:
This code segment is also used elsewhere, could you extract a function to avoid duplicating code?